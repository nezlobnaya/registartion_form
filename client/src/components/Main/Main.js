import { Route, Switch } from 'react-router-dom';
import Home from '../Home/Home';
import RecordsList from '../RecordsList/RecordsList';
import RecordSuccess from '../Success/RecordSuccess';
import RecordAdd from '../Form/Form';


function Main() {
    return(
      <Switch>
        <Route exact path="/" component={Home} />
        <Route exact path="/records" component={RecordsList} />
        <Route exact path="/records/new" component={RecordAdd} />
        <Route exact path="/success" component={RecordSuccess} />
      </Switch>
    );
  }

export default Main;