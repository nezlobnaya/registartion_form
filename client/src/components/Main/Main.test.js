import Main from './Main';
import { render, screen } from '@testing-library/react';


describe('Main', () => {
    test('Per Aspera Ad Astra!!', () => {
        render(<Main />);
        const linkElement = screen.getByText(/Per Aspera Ad Astra!!/i);
        expect(linkElement).toBeInTheDocument();
      });

      test('snapshot for App component with custom className and style', () => {  
        const { container } = render(<Main className="custom" style={{ background: 'red' }} />);
        expect(container).toMatchSnapshot();
      } );

});
