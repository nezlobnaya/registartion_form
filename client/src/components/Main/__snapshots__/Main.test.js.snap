// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot for App component with custom className and style 1`] = `
<div>
  <div
    class="jumbotron"
  >
    <h1>
      Per Aspera Ad Astra!!
    </h1>
  </div>
</div>
`;
